src/cli.rs

```diff
@@ -282,7 +286,11 @@ pub async fn main<H: Hooks, M: MigratorTrait>() -> eyre::Result<()> {
                 StartMode::ServerOnly
             };
 
-            let boot_result = create_app::<H, M>(start_mode, &environment).await?;
+            let b = binding.unwrap_or_else(|| "[::]".to_string());
+
+            let mut boot_result = create_app::<H, M>(start_mode, &environment).await?;
```

Comment:
I would suggest that you refrain from mutating`boot_result` directly.  I think we can improve the `environment. Load ()?` function (see at the top of the function) by add a struct called `config_overrides` for examples. when loading the configuration, we can override the YAML configuration with the struct values.

Reply:
One more note in this file, 
We have 2 `main` functions. One is when the feature `with-db` is on, and one is off. Please make sure that we are supporting the `binding` flag in both of the methods

Reply:
@kaplanelad, I have made all the requested changes save for one. That is a way to avoid mutating `boot_result` directly. Can you explain in more details how you think I should approach this. Are you suggesting that I should mutate the environment instead. Also, in `cli.rs` the `environment.load()` is called before matching the CLI command which means that I will need to override the environment before then should I want to avoid mutating the environment.

Reply:
@mikosco4real, the primary objective is to avoid using mutable references on the `boot_result`. To achieve this, we propose shifting the decision-making process for overrides to the `configuration` and refrain from modifying the configuration already loaded within the `boot_result` struct.

To implement this, I suggest introducing a new struct named `ConfigOverrides`. This struct will be initialized with variables from the CLI, specifically, in this case, with a `server_binding` option:
```rust
pub struct ConfigOverrides {
   server_binding: Option<String>,
}
```

During the parsing of CLI arguments, set the binding flag into an instance of this struct. Subsequently, when loading the environment, pass this struct as an argument to the configuration loading process to effect the necessary overrides:
```rust
let config_overrides = ConfigOverrides {
    server_binding: /* CLI binding value */,
};

let config = environment.with_overrides(&config_overrides)?;
```

This approach allows for a more structured and modular handling of configuration overrides without the need for mutability on the boot_result.